src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -748,4 +749,14 @@ public static void onCommandRegister(CommandDispatcher<CommandSource> dispatcher
         MinecraftForge.EVENT_BUS.post(evt);
         return evt;
     }
+
+    public static int onLivingBurnBySunlight(LivingEntity living, int duration)
+    {
+        LivingBurnBySunlightEvent event = new LivingBurnBySunlightEvent(living, duration);
+        MinecraftForge.EVENT_BUS.post(event);
+        duration = event.getResult() == Result.DEFAULT ? duration : event.getResult() == Result.ALLOW ? event.getDuration() : 0;
+        if (duration != 0)
+            living.setFire(duration);
+        return duration;
```

Comment:
This can be handled better imo. It's very messy for what it can be.

Maybe something like if the result isn't deny do logic.

Reply:
Changed the event from HasResult to Cancelable, this makes the factory cleaner.